### PR TITLE
Make dirname apply to hidden/tel/url/email inputs.

### PR DESCRIPTION
**Pull request by vinhill**
See https://github.com/whatwg/html/pull/9490

Dirname should soon apply to input elements of type hidden/tel/url/email in addition to text and search.

### CODE REVIEW

**Comment by bsmth** [comment]
Thanks a lot for the suggestions @vinhill - Is it correct that we're still waiting for implementations? We need [at least one browser to support this](https://developer.mozilla.org/en-US/docs/MDN/Writing_guidelines/What_we_write#when_we_document_a_new_technology) before we merge. I've added the appropriate label for now.

**Comment by vinhill** [comment]
@bsmth you're right. The spec just merged and I can add a note here once the change landed in Firefox.

File: files/en-us/web/html/element/input/index.md [hunk 1]
```diff
@@ -399,7 +399,7 @@ A few additional non-standard attributes are listed following the descriptions o
 
 - `dirname`
 
-  - : Valid for `text` and `search` input types only, the `dirname` attribute enables the submission of the directionality of the element. When included, the form control will submit with two name/value pairs: the first being the [`name`](#name) and [`value`](#value), the second being the value of the `dirname` as the name with the value of `ltr` or `rtl` being set by the browser.
+  - : Valid for `hidden`, `text`, `search`, `url`, `tel` and `email` input types, the `dirname` attribute enables the submission of the directionality of the element. When included, the form control will submit with two name/value pairs: the first being the [`name`](#name) and [`value`](#value), the second being the value of the `dirname` as the name with the value of `ltr` or `rtl` being set by the browser.
```
Comment (chrisdavidmills) [inline files/en-us/web/html/element/input/index.md:402]:
```suggestion
  - : Valid for `hidden`, `text`, `search`, `url`, `tel`, and `email` input types, the `dirname` attribute enables the submission of the directionality of the element. When included, the form control will submit with two name/value pairs: the first being the [`name`](#name) and [`value`](#value), and the second being the value of the `dirname` attribute as the name, with a value of `ltr` or `rtl` as set by the browser.
```

**Review by chrisdavidmills — APPROVED** [review]
@vinhill looks good, thanks! I've made a nitpicky suggestion on the attribute description, first to add a serial comma as per MDN style, and second to make some small changes to make the last sentence just a little bit easier to parse. 

I've approved this, and we can get it merged when the change actually drops in Firefox.

**Comment by vinhill** [comment]
The change landed, see https://github.com/mozilla/gecko-dev/commit/6bb941a5abbee3cd89e90b1889819e8fea2d98a1 and is in Nightly. @chrisdavidmills

**Comment by vinhill** [comment]
Sorry, I thought it was already in Nightly, but that is not the case yet.

**Comment by vinhill** [comment]
By now the change is applied to Nightly.

**Comment by bsmth** [comment]
Shall we also update https://developer.mozilla.org/en-US/docs/Web/HTML/Attributes#attribute_list with a description and a link to https://developer.mozilla.org/en-US/docs/Web/HTML/Attributes/dirname?

File: files/en-us/web/html/element/input/index.md [hunk 2]
```diff
@@ -399,7 +399,7 @@ A few additional non-standard attributes are listed following the descriptions o
 
 - `dirname`
 
-  - : Valid for `text` and `search` input types only, the `dirname` attribute enables the submission of the directionality of the element. When included, the form control will submit with two name/value pairs: the first being the [`name`](#name) and [`value`](#value), the second being the value of the `dirname` as the name with the value of `ltr` or `rtl` being set by the browser.
+  - : Valid for `hidden`, `text`, `search`, `url`, `tel`, and `email` input types, the `dirname` attribute enables the submission of the directionality of the element. When included, the form control will submit with two name/value pairs: the first being the [`name`](#name) and [`value`](#value), and the second being the value of the `dirname` attribute as the name, with a value of `ltr` or `rtl` as set by the browser.
```
Comment (bsmth) [inline files/en-us/web/html/element/input/index.md:402]: There's a page we can link to:

```suggestion
  - : Valid for `hidden`, `text`, `search`, `url`, `tel`, and `email` input types, the `dirname` attribute enables the submission of the directionality of the element. When included, the form control will submit with two name/value pairs: the first being the [`name`](#name) and [`value`](#value), and the second being the value of the `dirname` attribute as the name, with a value of `ltr` or `rtl` as set by the browser.
    For more information, see the [`dirname` attribute](/en-US/docs/Web/HTML/Attributes/dirname).
```

Comment (vinhill) [reply files/en-us/web/html/element/input/index.md:402]: Sounds good, though I think this should come after the example.

**Review by bsmth — APPROVED** [review]
Adding my +1 here.

See also:

- https://github.com/mdn/content/pull/28032
- https://github.com/mdn/content/pull/28296

**Pull request merged**